src/Servers/Kestrel/Core/src/Middleware/Internal/DuplexPipeStreamAdapter.cs

```diff
@@ -74,6 +80,8 @@ public override async ValueTask DisposeAsync()
                 // the application data to the underlying stream
                 await _outputTask;
             }
+
+            _disposed = true;
```

Comment:
It's unusual to set _disposed to true at the end of the method. The fact that it's unusual makes it seem like it's done for a reason, but if this method really could be called concurrently, we should probably be more thoughtful about it and lock.

Reply:
> It's unusual to set _disposed to true at the end of the method. 

I've seen both patterns, I'm fine with either, but I'll appease you for now :).

Reply:
I wouldn't have been concerned about it if there wasn't an await in the middle of the method.